docs/pipelines.md

```diff
@@ -760,6 +760,33 @@ spec:
           value: "someURL"
 ```
 
+#### Consuming `Task` execution results in `finally`
+
+Final tasks can be configured to consume `Results` of `PipelineTask` from the `tasks` section:
+
+```yaml
+spec:
+  tasks:
+    - name: clone-app-repo
+      taskRef:
+        name: git-clone
+  finally:
+    - name: discover-git-commit
+      params:
+        - name: commit
+          value: $(tasks.clone-app-repo.results.commit)
+```
+**Note:** The scheduling of such final task does not change, it will still be executed in parallel with other
+final tasks after all non-final tasks are done.
+
+The controller resolves task results before executing the finally task `discover-git-commit`. If the task
+`clone-app-repo` failed or skipped with [when expression](#guard-task-execution-using-whenexpressions) resulting in
+uninitialized task result `commit`, the finally Task `discover-git-commit` will be included in the list of
+`skippedTasks`. The controller logs the validation failure `unable to find result referenced by param "commit" in
+"discover-git-commit"` and continues executing rest of the final tasks. The pipeline exits with `completion` instead of
```

Comment:
note: i really dont think we should include anything about what the controller will likely log in these docs. if we start documenting that kind of thing, then users might come to rely on it.

imo the docs should cover the expected functionality that would be more or less protected by our backwards compatibility policy - if we wanted to change what the controller logs, i dont think we should need to change our docs.

i also think it's important that we distinguish between:
- how the features are implemented
- what the features do

users need to know about what the features do, and as much as possible shouldn't have to think about how the features are implemented (which also leaves us maintainers free to change the implementation around without impacting users)

Reply:
I can remove this log from here for now but without this log, if I am running into a finally task being part of `skippedTasks`, I have no way to look for `why` from doc perspective (without looking at the code).

Reply:
deleted log from doc.